vllm/model_executor/layers/fused_moe/fused_marlin_moe.py

```diff
@@ -196,27 +218,12 @@ def fused_marlin_moe(
         1] == w1.shape[1] * 16, "Hidden size mismatch w1"
     assert hidden_states.shape[1] == w2.shape[2] // (
         num_bits // 2), "Hidden size mismatch w2"
-    assert gating_output.shape[1] == w1.shape[0], "Number of experts mismatch"
```

Comment:
Why remove this?

Reply:
For expert parallelism, `gating_output.shape[1]` equals to global num experts, while `w1.shape[0]` equals to local num experts.